ingest/expression_files/dense_ingestor.py

```diff
@@ -44,62 +44,47 @@ def __init__(self, file_path, study_file_id, study_id, **kwargs):
 
     def matches_file_type(file_type):
         return file_type == 'dense'
 
-    # def validate_unique_header(self):
-    #     """Validates header has no duplicate values"""
-    #     if len(set(self.header)) != len(self.header):
-    #         self.error_logger.error(
-    #             "Duplicated header values are not allowed", extra=self.extra_log_params
-    #         )
-    #         return False
-    #     return True
-    #
-    # def validate_gene_keyword(self):
-    #     """Validates that 'Gene' is the first value in header"""
-    #     # File is an R formatted file
-    #     if (self.header[-1] == '') and (self.header[0].upper() != 'GENE'):
-    #         pass
-    #     else:
-    #         # If not R formatted file then first cell must be 'GENE'
-    #         if self.header[0].upper() != 'GENE':
-    #             self.error_logger.error(
-    #                 f'First cell in header must be GENE not {self.header[0]}',
-    #                 extra=self.extra_log_params,
-    #             )
-    #             return False
-    #     return True
-
-    # @trace
-    # def preprocess(self):
-    #     """Determines if file is R-formatted. Creates dataframe (df)"""
-    #     csv_file, open_file_object = self.open_file(self.file_path)
-    #     dtypes = {'GENE': object}
-    #
-    #     # # Remove white spaces and quotes
-    #     header = [col_name.strip().strip('\"') for col_name in self.header]
-    #     # See if R formatted file
-    #     if (header[-1] == '') and (header[0].upper() != 'GENE'):
-    #         header.insert(0, 'GENE')
-    #         # Although the last column in the header is empty, python treats it
-    #         # as an empty string,[ ..., ""]
-    #         header = header[0:-1]
-    #     else:
-    #         header[0] = header[0].upper()
-    #     # Set dtype for expression values to floats
-    #     dtypes.update({cell_name: 'float' for cell_name in header[1:]})
-    #     self.df = self.open_file(
-    #         self.file_path,
-    #         open_as='dataframe',
-    #         names=header,
-    #         skiprows=1,
-    #         dtype=dtypes,
-    #         # chunksize=100000, Save for when we chunk data
-    #     )[0]
+    def is_valid_format(self):
+        return all(
+            [
+                self.has_unique_header(),
+                self.has_gene_keyword(),
+            ]
+        )
+
+    def has_unique_header(self):
+        """Validates header has no duplicate values"""
+        if len(set(self.header)) != len(self.header):
+            self.error_logger.error(
+                "Duplicate header values are not allowed", extra=self.extra_log_params
+            )
+            return False
+        return True
+
+    def has_gene_keyword(self):
+        """Validates that 'Gene' is the first value in header"""
+        # File is an R formatted file
+        if (self.header[-1] == '') and (self.header[0].upper() != 'GENE'):
```

Comment:
In `R`, if a user outputs a CSV using `write.table()`, what you get is a header line that is one entry shorter than each successive line.  Specifically - there is no padding for the `GENE` keyword.  That's what we want to check for here.

Reply:
Fixed in 15dc741